src/parser/mod.rs

```diff
@@ -8382,6 +8382,10 @@ impl<'a> Parser<'a> {
                     Ok(DataType::Tuple(field_defs))
                 }
                 Keyword::TRIGGER => Ok(DataType::Trigger),
+                Keyword::ANY => {
```

Comment:
```suggestion
                Keyword::ANY if self.peek_keyword(Keyword::TYPE) => {
```
Thinking we can add this guard to ensure that it doesn't conflict with a custom data type that happens to be called `ANY`?

Reply:
Ah yes that is much more in line with the desired / intended behavior. As the current approach would result in a parsing error should we come across an ANY type I think.

Reply:
I tried implementing this change but it seems to break. hmm

Reply:
As this is the first time I'm touching the codebase, I needed to understand the general mechanisms. Your change worked in conjunction with `let _ = self.parse_keyword(Keyword::TYPE);`, which makes intitutive sense after looking at the code in general.